pilot/pkg/networking/grpcgen/lds.go

```diff
@@ -147,7 +148,7 @@ func buildFilterChain(nameSuffix string, tlsContext *tls.DownstreamTlsContext) *
 			Name: "inbound-hcm" + nameSuffix,
 			ConfigType: &listener.Filter_TypedConfig{
 				TypedConfig: util.MessageToAny(&hcm.HttpConnectionManager{
-					// TODO gRPC doesn't support httpfilter yet; sending won't cause a NACK but they don't do anything
+					HttpFilters: []*hcm.HttpFilter{xdsfilters.Router},
```

Comment:
Is there any reason we choose to set this? Since we don't have other filters, it shouldn't matter, but it seems like this just acts like a terminator to stop processing filters:

https://github.com/grpc/grpc-go/blob/ef66d13abb84ad6c6d99c8cbf3697607b7891f32/xds/internal/resolver/serviceconfig.go#L263-L266

Reply:
they may it required: https://github.com/grpc/grpc-go/blame/458ea7640a92039aad37edc67b63e6d040a93320/xds/internal/xdsclient/xds.go#L260